libbeat/processors/add_process_metadata/add_process_metadata.go

```diff
@@ -93,18 +101,44 @@ func newProcessMetadataProcessorWithProvider(cfg *common.Config, provider proces
 		return nil, errors.Wrapf(err, "fail to unpack the %v configuration", processorName)
 	}
 
-	p := addProcessMetadata{
-		config:   config,
-		provider: provider,
-		log:      log,
-	}
-	if p.mappings, err = config.getMappings(); err != nil {
+	mappings, err := config.getMappings()
+
+	if err != nil {
 		return nil, errors.Wrapf(err, "error unpacking %v.target_fields", processorName)
 	}
 
+	var p addProcessMetadata
+
+	// don't use cgroup.ProcessCgroupPaths to save it from doing the work when container id disabled
+	if ok := containsValue(mappings, "container.id"); ok {
```

Comment:
I'd suggest using built in functions to check keys.
https://github.com/elastic/beats/blob/78e481d541796ad572ec28d0f4324701fc54aec7/libbeat/common/mapstr.go#L161

Reply:
In this case, it is checking if the value exist or not instead of the key.